src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.Cache.cs

```diff
@@ -98,7 +106,7 @@ public bool Equals(CacheKey other)
 
             return
                 Flags == other.Flags &&
-                Settings.Equals(other.Settings) &&
+                SettingsSpan.SequenceEqual(other.SettingsSpan) &&
```

Comment:
Can you add a test pls?

Reply:
What sort of test do you have in mind?

Reply:
I mean the simplest unit test that equal is still equal :D 
see comment above:

> I think there might be some padding, e.g. between MtuDiscoveryMissingProbeCount and DestCidUpdateIdleTimeoutMs it seems like there'd be 3 bytes of padding - which would make the SequenceEquals approach incorrect.

Reply:
Specifically, the approach using `SequenceEquals` would be incorrect if the padding bytes are different in each, which should compare to equal, but wouldn't.

Reply:
As discussed with @rzikm offline: we should consider updating MsQuic generator code to include an additional field-by-field equality comparison to QUIC_SETTINGS (i.e. approach from  https://github.com/dotnet/runtime/pull/101133#discussion_r1568624284). Adding it to generator will also prevent losing any newly-added fields to the struct.